Engine_Revit_UI/Query/ElementIds/ElementIdsByCategoryNames.cs

```diff
@@ -0,0 +1,82 @@
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+using Autodesk.Revit.DB;
+using Autodesk.Revit.UI;
+
+using BH.oM.Base;
+using BH.Engine.Adapters.Revit;
+using BH.oM.Adapters.Revit;
+using BH.oM.Adapters.Revit.Enums;
+using BH.oM.Adapters.Revit.Interface;
+using BH.oM.Data.Requests;
+using BH.oM.Reflection.Attributes;
+
+
+
+namespace BH.UI.Revit.Engine
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Get Elements as ElementIds by Category name")]
+        [Input("document", "Revit Document where ElementIds are collected")]
+        [Input("categoryName", "List of Category names to be used as filter")]
+        [Input("ids", "Optional, allows the filter to narrow the search from an existing enumerator")]
+        [Output("elementIdsByCategoryNames", "An enumerator for easy iteration of ElementIds collected")]
+        public static IEnumerable<ElementId> ElementIdsByCategoryNames(this Document document, List<string> categoryNames, IEnumerable<ElementId> ids = null)
+        {
+            List<ElementId> returned = new List<ElementId>();
+
+            if (document == null || categoryNames.Count == 0)
+                return null;
+
+            foreach(string name in categoryNames)
+            {
+                BuiltInCategory builtInCategory = document.BuiltInCategory(name);
+                if (builtInCategory == Autodesk.Revit.DB.BuiltInCategory.INVALID)
+                    BH.Engine.Reflection.Compute.RecordError("Couldn't find a Category named " + name + ".");
+                
+                FilteredElementCollector collector = ids == null ? new FilteredElementCollector(document) : new FilteredElementCollector(document, ids.ToList());
```

Comment:
If the collector is outside the loop in the first iteration of a category name it'll filter down to that category, no allowing for the next iteration of names. Unless FilteredElementCollector will reset itself when queried again?

Reply:
Huh, good point. `OfCategory` returns `FilteredElementCollector` instead of `void`, so I would expect a new one to be created. Are you sure that is not the case?

Reply:
Yep just tested, if outside the loop `FilteredElementCollector` will be set only to the first `OfCategory`. Left within loop for now.